lib/ansible/modules/windows/win_share.ps1

```diff
@@ -149,32 +158,53 @@ Try {
 
         # need to (re-)create share
         If (-not $share) {
-            New-SmbShare -Name $name -Path $path
-            $share = Get-SmbShare $name -ErrorAction SilentlyContinue
+            if (-not $check_mode) {
+                New-SmbShare -Name $name -Path $path
+            }
+            $share = Get-SmbShare -Name $name -ErrorAction SilentlyContinue
 
             $result.changed = $true
+            $result.actions += "New-SmbShare -Name $name -Path $path"
         }
         If ($share.Path -ne $path) {
-            Remove-SmbShare -Force -Name $name
-
-            New-SmbShare -Name $name -Path $path
-            $share = Get-SmbShare $name -ErrorAction SilentlyContinue
-
+            if (-not $check_mode) {
+                Remove-SmbShare -Force -Name $name
```

Comment:
`New-SmbShare`, `Set-SmbShare` and `Remove-SmbShare` has `-WhatIf` support, so we could do `-WhatIf:$check_mode` instead.